### ISSUE TITLE
No way to massively remove unused imports

### ISSUE THREAD

**Issue by kkolyan** [intellij-rust/intellij-rust#8924]
## Environment

* **IntelliJ Rust plugin version:** 0.4.171.4656-213
* **Rust toolchain version:** 1.63.0-nightly (420c970cb 2022-06-09) x86_64-pc-windows-msvc
* **IDE name and version:** CLion 2021.3.4 (CL-213.7172.20)
* **Operating system:** Windows 10 10.0
* **Macro expansion engine:** new
* **Name resolution engine:** new
* **Additional experimental features:** org.rust.macros.proc

## Problem description
At the active development phase that's very frustration to have a lot of "unused import" warnings. And as I see, It's IDE level inspection works fairly well. Why do not allow user to remove them all by one hotkey? For example by "Optimize imports".
Or this feature:
![image](https://user-images.githubusercontent.com/3194236/173192932-38f64f83-7536-4011-a267-7065ce2395af.png)
It explicitly should do what I need, but it does nothing (I mean option to fix all in file).

Ok, I've googled a lot of discussions that "unused imports" inspection is unreliable because has false-positives, but why do not allow this as an experimental feature? I'm ok to fix rare false-positives manually.

So I'm creating new issue as suggested [here](https://github.com/intellij-rust/intellij-rust/pull/7495#issuecomment-1028833990)

`org.rust.macros.proc` toggle doesn't make any difference.

## Steps to reproduce
1. Create fresh binary executable project with following code
```
use std::borrow::Borrow;
use std::ops::Div;

fn main() {
    println!("Hello, world!");
}
```
2. Optimize imports with `Ctrl+Alt+O` (unused imports left)
3. Try to remove them with the `Alt+Enter/External linter: remove whole item` - it works (for a single item)
4. Try to apply fix to whole file with `Alt+Enter/External linter: remove whole item/Fix all 'Annotator' problems in file` - it does nothing

(`Language & Frameworks/Rust/External Linters/External Linter` is default - "Cargo Check")

**Comment by peterwilli** [comment]
I still have this problem today. I wonder if something can be done about it?

**Comment by dima74** [comment]
@kkolyan @peterwilli it can be done using builtin `Import optimizer` (Ctrl+Alt+O), just [enable `Unused imports` inspection](https://user-images.githubusercontent.com/6505554/144234045-f94ad472-4196-4194-8374-f60d8b3c90fc.png) in `Editor | Inspections`

**Comment by peterwilli** [comment]
> @kkolyan @peterwilli it can be done using builtin `Import optimizer` (Ctrl+Alt+O), just [enable `Unused import` inspection](https://user-images.githubusercontent.com/6505554/144234045-f94ad472-4196-4194-8374-f60d8b3c90fc.png) in `Editor | Inspections`

Hey Dima,

I tried this, but it still doesn't remove any unused imports. 
![image](https://user-images.githubusercontent.com/1212814/197356041-bf12230c-9790-4579-8d3c-d22ae13366d2.png)

My IntelliJ version:
![image](https://user-images.githubusercontent.com/1212814/197356069-16d4bcc8-fc95-4887-8152-f113227a026a.png)

**Comment by dima74** [comment]
@kkolyan @peterwilli Oh, sorry, I forgot, could you please also enable `org.rust.cargo.evaluate.build.scripts` and `org.rust.macros.proc` [experimental features](https://plugins.jetbrains.com/plugin/8182-rust/docs/rust-faq.html#experimental-features)

**Comment by dima74** [comment]
@kkolyan this is very strange. Which version of the plugin are you using? Could you please attach your [`idea.log`](https://intellij-support.jetbrains.com/hc/en-us/articles/207241085-Locating-IDE-log-files)? Could you also try to disable ["Enable inspection only if procedural macros are enabled"](https://user-images.githubusercontent.com/6505554/197757415-6f7b13c1-d53a-4df0-b0f8-f40b56ec5c83.png) option in intention settings?

> "Unused import" is reported

By the way it is reported by `cargo clippy` external linter, not by our builtin annotator. Import will be fully grayed out when reported by our annotator:
<img src="https://user-images.githubusercontent.com/6505554/197756868-06a406da-2eff-4186-80c1-9f5f1ba9f68c.png" width="235">

**Comment by peterwilli** [comment]
After enabling the experimental features and disabling ["Enable inspection only if procedural macros are enabled"](https://user-images.githubusercontent.com/6505554/197757415-6f7b13c1-d53a-4df0-b0f8-f40b56ec5c83.png), all works, but only in some cases.

For some reason, this doesn't work: 
![image](https://user-images.githubusercontent.com/1212814/202469076-dfaedd7a-a744-4e68-985e-df0324fd3328.png)

But this does: 
![image](https://user-images.githubusercontent.com/1212814/202469161-d65a7a10-89fc-43e5-a6d8-1d3a704745ee.png)

**Comment by ccleve** [comment]
Can we reopen this issue? I've followed all the suggestions above and the system does not remove unused imports.

**Comment by dima74** [comment]
@ccleve could you please create a new issue from the IDE (via `Help | Create New Issue`), and provide more information (are imports highlighted as unused but not removed, or not highlighted, etc)

**Comment by ccleve** [comment]
@dima74 

I did a full factory reset on settings using File / Manage IDE Settings / Restore Default Settings..., followed by unchecking Settings / Editor / Inspections / Rust / Lints / Unused import / "Enable inspection only if procedural macros are enabled". I right-clicked on the src dir, hit Optimize Imports, and it worked.

**Comment by xring** [comment]
@ccleve By unchecking `Enable inspection only if procedural macros are enabled`, it worked. Thanks.

**Comment by ccleve** [comment]
Let's reopen this. The solution above does not work with RustRover. It removed about 25% of the unused imports.

**Comment by valydumitru01** [comment]
The problem persists for RustRover 2024.3.6

**Comment by mati865** [comment]
This bug tracker is no longer used, as per README.md you should report to JetBrain's YouTrack.
